src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -463,7 +464,7 @@ public void StopStake()
 
             BlockTemplate blockTemplate = null;
 
-            while (!this.stakeCancellationTokenSource.Token.IsCancellationRequested)
+            while (this.stakeCancellationTokenSource != null && !this.stakeCancellationTokenSource.Token.IsCancellationRequested)
```

Comment:
This is not correct. The correct solution is to wait for this method to complete before setting the cancellation source to null.

Reply:
Yes, that would be cleaner - working on it...

Reply:
So now you can remove it